### ISSUE TITLE
Add different types of the same size wheels.

### ISSUE THREAD

**Issue by Vasyan2006** [CleverRaven/Cataclysm-DDA#27612]
**Is your feature request related to a problem? Please describe.**  
Lets compare offroad coefficients for different vehicles:
Sports Car: 53%
Car: 54%
SUV: 44%
They all have the same wheels but SUV is the heavy one, so it has worst behavior. But considering vehicle names these coefficients should be more like 25-50-80.
Same for 11" motorbike wheels, for example. We have Motorcycle, Quad Bike and Superbike. And Quad Bike only better because it has more wheels.

**Describe the solution you'd like**  
Most wheel sizes should have 3 different flavors (item flags?): normal, offroad and onroad.
Offroad wheels have worse overall performance, but very high offroad capacity. Onroad wheels are the opposite, very good on roads but useless outside. 
Actual irl performance also heavily depends on transmissions and suspensions, but they are not in the game and can be ignored for now.
This will make vehicles more diverse and add more options for players to build new self-propelled carts.

**Additional context**  
It is related to #27520, because using only wheel mass and area is not enough sometimes.

**Comment by cheesealmighty** [comment]
This might be different in US, but here, usually apart from wheel sizes, most wheels you'll find are either winter or summer wheels (and all season, but all season tires are garbage). Of course they provide different traction, but  "Onroad" wheels are strictly for racetracks and wouldn't really be findable anywhere else. 

This also goes for SUVs, as they use the same tires as other cars, as they are designed to be used in cities nowadays. 

There could rarely be offroad vehicle spawns, with appropriate tires like Suzuki, Land Rover, Humvee etc. Although realistically, these offroad tires wouldn't really be that bad on the road either. So this could be presented as something to be sought out for. 

Basically, Winter-Summer-Offroad tires might a better naming system overall. Huh. I suppose this is not much different from what you're suggesting, apart from naming.

**Comment by Vasyan2006** [comment]
By different wheels I mean not only tires, but entire suspension block, because it is magically appear is we install a wheel on any tile. And all other things which create the difference between racing and off-road vehicles.
So sport car will not be always better compare to SUV.

**Comment by GearheadLydia** [comment]
> [...] entire suspension block, because it is magically appear if we install a wheel on any tile.

This, combined with the "road rollers are OP" issue raised elsewhere, suggests that it might be worth trying to look at various suspension configurations to attach to the car before attaching the wheels. Might also affect braking.

**Comment by kevingranade** [comment]
I'm in no way against breaking wheels up into wheel/axle/suspension/other?, there just hasn't been a clear reason to do so.

**Comment by GearheadLydia** [comment]
For simplicity, I'm going to leave brakes out of this because generally one wants them as good as possible, so each successive option would basically be a straight upgrade over the next.

Personally, I think that the best way to split up suspensions would be:

- zero travel suspension (road rollers) with essentially no offroad and poor road handling
- road-oriented suspension that results in fewer losses of control on pavement, but worse offroad
- offroad-oriented suspension that gives great offroad, at the cost of some loss of control on road

**Comment by Vasyan2006** [comment]
Imho separating wheels into sport and offroad groups can be done only by adding item flags, without adding new entities.

Adding suspensions can be interesting gameplay mechanics it will work similar to swapable battery case. Changing a wheel irl is a simple task, but installing new wheel on a vehicle is much more complicated. 
So a wheel can be installed only over suspension part with minimal skills. But installing new "swapable  wheel case" will require much more skills, tools and materials.

**Comment by mlangsdorf** [comment]
Currently, traction is handled by (pseudo-code follows)
```
traction_wheel_area = 0;
total_wheel_area = 0;
for each wheel:
    wheel area = wheel width * wheel diameter
    total_wheel_area += wheel area
    move_mod = move_cost_tern_furn( pp );
    move_mod += 4 if terrain isn't "FLAT" or +2 if terrain isn't "ROAD"
    traction_wheel_area += flat, road move cost * wheel_area / move_mod
mass_penalty = vehicle_mass_in_kg * ( 1 - traction_wheel_area / wheel_area )
traction = the lower of 1 or wheel_traction_area / mass_penalty to a minimum of 0.1
vehicle max velocity = nominal max * traction
vehicle acceleration = nominal acceleration * traction
```
One relatively easy change would be adding a `terrain_modifiers` array to each wheel vehicle_part's JSON definition:
`"terrain_modifiers": [ [ "T_FLAG1", val1 ], [ "T_FLAG2", val2 ], ... ],`
so you could have off-road wheels that are less penalized for not being on "ROAD", or road rollers that are much more penalized for not being on "FLAT" than other wheels, or eventually snow skis that are harshly penalized for not being on "SNOW" (not that we have a SNOW flag yet).

Suggested values for the different types of wheels:
```
"road_roller: { "terrain_modifiers": [ [ "FLAT", 6 ], [ "ROAD", 3 ] ] },
"standard": { "terrain_modifiers": [ [ "FLAT", 4 ], [ "ROAD", 2 ] ] },
"racing": { "terrain_modifiers": [ [ "FLAT", 5 ], [ "ROAD", 2 ] ] },
"off-road": { "terrain_modifiers": [ [ "FLAT", 3 ], [ "ROAD", 1 ] ] }
"caterpillar_track": { "terrain_modifiers": [ [ "FLAT", 3 ] ] }
```

**Comment by Vasyan2006** [comment]
Road rollers still will be the best road wheels due to largest area.
We need something like "overall road performance" coefficient which is multiplied by wheel area:
```
"road_roller": 0.2,
"standard": 1.0
"racing": 1.25
"off-road": 0.8
"caterpillar_track": 0.5
```
Caterpillar tracks will be worse on road compare to standard wheels, but they will get less penalty on dirt and higher final offroad coefficient.

**Comment by mlangsdorf** [comment]
Instead of adding another value to the JSON to help calculate wheel area, I think it would be easier to change `vehicle_part::wheel_area()` to just take a value from JSON.  Currently it's calculated from the underlying item's diameter and width, and we could use that as a basis but use new values:


Wheel | item_id | part_id | Diameter | Width | New Area | Terrain Mods
-- | -- | -- | -- | -- | -- | -- 
Caster Set | wheel_caster | | 4 | 1 | 4 | "road_roller"
Off-road Bicycle Wheel* | wheel_bike_or | wheel_bike_or | 10 | 2 | 18 | "off-road"
Unicycle/Bicycle Wheel | wheel_bicycle | wheel_bicycle | 10 | 2 | 20 | "standard"
Small Wheel | wheel_small | wheel_small | 10 | 3 | 30 | "racing"
Wheelbarrow wheel | wheel_barrow | wheel_barrow | 8 | 3 | 30 | "off-road"
Wheelchair Wheel | wheel_wheelchair | wheel_wheelchair | 20 | 2 | 40 | "racing"
Off-road motorbike Wheel* | wheel_motorbike_or | wheel_motorbike_or | 11 | 4 | 40 | "off-road"
Motorcycle Wheel | wheel_motorbike | wheel_motorbike | 11 | 4 | 44 | "standard"
Wood Wheel | wheel_wood | wheel_wood | 20 | 3 | 60 | "road_roller"
Metal Wheel | wheel_metal | wheel_metal | 20 | 4 | 80 | "road_roller"
Car Wheel | wheel | wheel | 17 | 9 | 153 | "standard"
Off-road Wide Wheel* | wheel_wide_or | wheel_wide_or | 24 | 14 | 320 | "off-road"
Wide Wheel | wheel_wide | wheel_wide | 24 | 14 | 360 | "standard"
Racing Slick* | wheel_slick | wheel_slick | 24 | 14 | 400 | "racing"
Roller Drum | hd_steel_drum | roller_drum | 60 | 60 | 400 | "road_roller"
Armored Wheel | wheel_armor | wheel_armor | 32 | 20 | 480 | "standard"
Rubber Treads** | tread1 | tread1 | 30 | 15 | 900 | "caterpillar_track"
Steel Treads** | tread2 | tread2 | 50 | 25 | 2500 | "caterpillar_track"
HD Steel Treads** | tread3 | tread3 | 60 | 30 | 3600 | "caterpillar_track"

Items marked with `*` are new items
* Off-road bike wheels would be the slightly wider tires found on dirt bikes, with slightly less effective contact area but somewhat better off-road performance
* Off-road wide wheels would be the kind of knobbed wheels seen on construction vehicles
* Racing Slicks are F1 style, wide smooth wheels that would have severe penalties on wet surfaces due to the lack of treading and quicker hydroplaning.  They don't handle rough terrain well either.

Items marked with `**` are taken from blazemod but are roughly where caterpillar track is going to be eventually.

So for high end vehicle construction, you have a couple of options:
* Roller drums are insanely heavy (1600+ lbs per drum), nigh invulnerable to damage, but have relatively little wheel area and terrible off-road performance.  They have very low rolling resistance, so speed isn't penalized too heavily other than the massive weight.  They're also "slicks" and would take severe penalties for water on the road when that's implemented.
* Armored wheels are fairly light (~60 lbs per wheel), resistant to damage, decent wheel area, decent off-road performance, and reasonably low rolling-resistance. 
* Treads are heavy ( 250-750 lbs per segment), resistant to damage, and excellent wheel area and off-road performance.  They have terrible rolling resistance, so you're getting that great off-road performance at the cost of overall speed.

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/tips-tricks-and-newb-questions-reborn/16870/1884
